pkg/catalog/manager/traverse.go

```diff
@@ -54,35 +90,39 @@ func (m *Manager) traverseAndUpdate(helm *helmlib.Helm, commit string, cmt *Cata
 		}
 	}
 
-	entriesWithErrors, entriesToProcess := m.preprocessCatalog(catalog, index.IndexFile.Entries)
-	for chart, metadata := range entriesToProcess {
-		newHelmVersionCommits[chart] = v32.VersionCommits{
-			Value: map[string]string{},
+	catalogTemplateMap, err := m.getTemplateMap(catalogName, templateNamespace)
+	if err != nil {
+		return err
+	}
+
+	// Check if a version has been deleted from a template or if an existing template version has changed.
+	hasChartChanged := func(template *v3.CatalogTemplate, metadata helmlib.ChartVersions) bool {
+		if template == nil || metadata == nil {
```

Comment:
We should return true here? If a version exist in metadata but not in template, the chart has changed(an entry has been deleted)

Reply:
This is a good point. I will upgrade this logic for that scenario.

Reply:
The length check covers your case, but this line should be `&&` instead of `||`.